compiler/modules/front/expr.bal

```diff
@@ -931,15 +932,15 @@ function codeGenMappingConstructor(ExprContext cx, bir:BasicBlock bb, t:SemType?
         }
         fieldPos[name] = f.startPos;
         if mat.names.indexOf(name) == () {
-            if mat.rest == t:NEVER {
+            if t:cellDeref(tc, mat.rest) == t:NEVER {
```

Comment:
Not just for the mapping `rest`, There are several places we need to basically get the `SemType`(of BT_CELL) and return its deref type. So added the function `cellDeref()` and used it for mapping atom rest type deref instead of having a separate function for rest alone.